### DISCUSSION TITLE
Making sense of Native Land API

### DISCUSSION THREAD

**Discussion by byandell** [earthlab-education/Earth-Analytics-AY24#237] [General]
The site <https://native-land.ca/resources/api-docs> provides access to geographic information about Tribes around the world. However, the API is pretty basic, and it is difficult to figure out how to turn the URL into a geoJSON or shapefile. Ideas welcome.

**Comment by eculler** [comment]
Hi @byandell, it looks like an API search should give you a GeoJSON. Can you provide an example of what you tried and what output you got?

It is possible, but can sometimes be convoluted, to convert between geoJSON and python geometry types or GeoDataFrames.

If you want, you can work in the first-map Codespace to get one of the polygons from the Native Lands API instead of from Open Street Map. That way it's also easy for us to try out the code you're using and poke around. You may need to use the `requests` library in Python to get results, since I don't think there are Python bindings for that API like the osmnx package. I'd recommend feeding the API documentation into ChatGPT and asking for sample code as a starting point.

**Comment by byandell** [comment]
Elsa,
Sounds good. I will work this up over the next week.
Brian

brian.yandell @ wisc.edu
byandell.github.io<https://byandell.github.io/>
calendar<https://go.wisc.edu/gpo59a>

From: Elsa Culler ***@***.***>
Date: Thursday, September 5, 2024 at 5:02 PM
To: earthlab-education/Earth-Analytics-AY24 ***@***.***>
Cc: Brian Yandell ***@***.***>, Mention ***@***.***>
Subject: Re: [earthlab-education/Earth-Analytics-AY24] Making sense of Native Land API (Discussion #237)

Hi @byandell<https://urldefense.com/v3/__https:/github.com/byandell__;!!Mak6IKo!NGwmiMds8q9UTIE9pH3XvH-V9mXW-CELI9RQ_XaKytTt8BGYBHETkAZVwhkFb51p3fbyR7n_B-9f0cYz_HR-r9MJrg$>, it looks like an API search should give you a GeoJSON. Can you provide an example of what you tried and what output you got?

It is possible, but can sometimes be convoluted, to convert between geoJSON and python geometry types or GeoDataFrames.

If you want, you can work in the first-map Codespace to get one of the polygons from the Native Lands API instead of from Open Street Map. That way it's also easy for us to try out the code you're using and poke around. You may need to use the requests library in Python to get results, since I don't think there are Python bindings for that API like the osmnx package. I'd recommend feeding the API documentation into ChatGPT and asking for sample code as a starting point.

**Reply by byandell** [reply]
The following works:
https://github.com/byandell-Tribal/first-map-template/blob/main/native-land.ipynb
which is great. This is based on code from Nate Q shared with OLC students in ESIIL Stars, and they shared with me (via Dana G).
The repo
https://github.com/byandell-Tribal/native-land
is my working space for this. You will see [native-land.py](https://github.com/byandell-Tribal/native-land/blob/main/native-land.py), which is essentially the above-cited code. In addition [native-land.R](https://github.com/byandell-Tribal/native-land/blob/main/native-land.R) is a first attempt at R, using https://www.codeconvert.ai/python-to-r-converter to try to convert from python to R. It is able to GET the list from the API, but I have had no success turning this list object into a geojson object, either on my own or with packages like geojsonio or geojsonsf. I am sure I am missing something simple.
Cheers, Brian
